client/cli/src/config.rs

```diff
@@ -0,0 +1,308 @@
+	fn get_wasm_method(&self) -> Result<WasmExecutionMethod> {
+		Ok(Default::default())
+	}
+
+	/// Get the execution strategies
+	fn get_execution_strategies(&self, _is_dev: bool) -> Result<ExecutionStrategies> {
+		Ok(Default::default())
+	}
+
+	/// Get the RPC HTTP address (`None` if disabled)
+	fn get_rpc_http(&self) -> Result<Option<SocketAddr>> {
+		Ok(Default::default())
+	}
+
+	/// Get the RPC websocket address (`None` if disabled)
+	fn get_rpc_ws(&self) -> Result<Option<SocketAddr>> {
+		Ok(Default::default())
+	}
+
+	/// Get the RPC websockets maximum connections (`None` if unlimited)
+	fn get_rpc_ws_max_connections(&self) -> Result<Option<usize>> {
+		Ok(Default::default())
+	}
+
+	/// Get the RPC cors (`None` if disabled)
+	fn get_rpc_cors(&self, _is_dev: bool) -> Result<Option<Vec<String>>> {
+		Ok(Some(Vec::new()))
+	}
+
+	/// Get the prometheus configuration (`None` if disabled)
+	fn get_prometheus_config(&self) -> Result<Option<PrometheusConfig>> {
+		Ok(Default::default())
+	}
+
+	/// Get the telemetry endpoints (if any)
+	fn get_telemetry_endpoints(
+		&self,
+		chain_spec: &Box<dyn ChainSpec>,
+	) -> Result<Option<TelemetryEndpoints>> {
+		Ok(chain_spec.telemetry_endpoints().clone())
+	}
+
+	/// Get the telemetry external transport
+	fn get_telemetry_external_transport(&self) -> Result<Option<ExtTransport>> {
+		Ok(Default::default())
+	}
+
+	/// Get the default value for heap pages
+	fn get_default_heap_pages(&self) -> Result<Option<u64>> {
+		Ok(Default::default())
+	}
+
+	/// Returns `Ok(true)` if offchain worker should be used
+	fn get_offchain_worker(&self, _roles: Roles) -> Result<bool> {
+		Ok(Default::default())
+	}
+
+	/// Get sentry mode (i.e. act as an authority but **never** actively participate)
+	fn get_sentry_mode(&self) -> Result<bool> {
+		Ok(Default::default())
+	}
+
+	/// Returns `Ok(true)` if authoring should be forced
+	fn get_force_authoring(&self) -> Result<bool> {
+		Ok(Default::default())
+	}
+
+	/// Returns `Ok(true)` if grandpa should be disabled
+	fn get_disable_grandpa(&self) -> Result<bool> {
+		Ok(Default::default())
+	}
+
+	/// Get the development key seed from the current object
+	fn get_dev_key_seed(&self, _is_dev: bool) -> Result<Option<String>> {
+		Ok(Default::default())
+	}
+
+	/// Get the tracing targets from the current object (if any)
+	fn get_tracing_targets(&self) -> Result<Option<String>> {
+		Ok(Default::default())
+	}
+
+	/// Get the TracingReceiver value from the current object
+	fn get_tracing_receiver(&self) -> Result<sc_tracing::TracingReceiver> {
+		Ok(Default::default())
+	}
+
+	/// Get the node key from the current object
+	fn get_node_key(&self, _net_config_dir: &PathBuf) -> Result<NodeKeyConfig> {
+		Ok(Default::default())
+	}
+
+	/// Get maximum runtime instances
+	fn get_max_runtime_instances(&self) -> Result<Option<usize>> {
+		Ok(Default::default())
+	}
+
+	/// Create a Configuration object from the current object
+	fn create_configuration<C: SubstrateCLI>(
```

Comment:
tbh this is a really great change, the `update_config` _methodology_ was very weird.

Reply:
Yes I think it did its job but this way is better even though more verbose.... Thanks!! Took me sometime to come up with this idea